moebot_bot/bot/commands/timer.go

```diff
@@ -0,0 +1,71 @@
+package commands
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/camd67/moebot/moebot_bot/bot/permissions"
+	"github.com/camd67/moebot/moebot_bot/util"
+	"github.com/camd67/moebot/moebot_bot/util/db"
+)
+
+type TimerCommand struct {
+	chTimers util.SyncChannelTimerMap
+	Checker  permissions.PermissionChecker
+}
+
+func NewTimerCommand() *TimerCommand {
+	tc := &TimerCommand{}
+	tc.chTimers = util.SyncChannelTimerMap{
+		RWMutex: sync.RWMutex{},
+		M:       make(map[string]time.Time),
+	}
+	return tc
+}
+
+func (tc *TimerCommand) Execute(pack *CommPackage) {
+	channelID := pack.message.ChannelID
+	if len(pack.params) > 0 && strings.EqualFold(pack.params[0], "start") {
+		if tc.Checker.HasPermission(pack.message.Author.ID, pack.member.Roles, pack.guild, db.PermMod) {
+			tc.chTimers.Lock()
+			tc.chTimers.M[channelID] = time.Now()
+			tc.chTimers.Unlock()
+			pack.session.ChannelMessageSend(pack.message.ChannelID, "Timer started! OwO")
+		} else {
+			pack.session.ChannelMessageSend(pack.message.ChannelID, pack.message.Author.Mention()+", you... you don't have permission to do that! ( ≧Д≦)")
+		}
+	} else {
+		tc.chTimers.Lock()
```

Comment:
For this one we can just do an `RLock` and `RUnlock` instead so it doesn't block other reads. `Lock` is only necessary when writing.